pageserver/src/tenant/timeline.rs

```diff
@@ -354,6 +354,17 @@ macro_rules! try_no_ondemand_download {
     }};
 }
 
+#[macro_export]
+macro_rules! try_page_reconstruct_result {
```

Comment:
I don't understand this macro. When should it be used? How is it different from just `?` ?

Reply:
`?` only works on `Result` and `Option`. It's not possible to implement it for custom types because `std::ops::Try` is nightly.

Reply:
Ah, I missed that `PageReconstructResult` is not just an alias for `std::result:Result<T, PageReconstructError>`. Gotcha

Reply:
a comment would be nice